pkg/aws/aws.go

```diff
@@ -89,9 +89,15 @@ func DefaultProfile() string {
 }
 
 func (ac *awsCloud) OpenPorts(ports []api.PortSpec, status reporter.Interface) error {
-	status.Start(messageRetrieveVPCID)
 	defer status.End()
 
+	if len(ports) == 0 {
+		status.Warning("Ignoring attempt to open ports with no ports given")
```

Comment:
Warnings shown to the user should be actionable; this isn’t something the user can fix.

Reply:
Hence I wanted it to return an error, but there was pushback against it even thought it is an unintended behavior.

This was a compromise I was trying to suggest, what do you suggest?

Reply:
I suggest returning directly:

```suggestion
```

Reply:
Calling "open ports" with no ports is probably some bug on the client, hence I think it would be better to either warn or even error out in such a case.

I prefer not hiding possible bugs by silently succeeding.

Reply:
> I prefer not hiding possible bugs by silently succeeding.

I understand your point but it's not necessarily a bug on the client, in fact I'd say it probably isn't in this case. As I've mentioned before, there's many APIs out there that silently handle the trivial case of no input/nothing to do, in fact it's usually a convenience for callers.  That said this isn't a widely-used API so I don't really think it matters either way so I'm not going to engage in a battle 😄

Reply:
My main point is that this isn’t great UX: we foist the warning on the user, even though they can’t do anything about it. What do users do when they get a warning they don’t understand? We can hope they’ll ignore it if everything works ultimately, but if that’s what we’re hoping for, we might as well not show the warning.

As far as this being a bug, my preference is to handle the empty slice in one place (here), and get rid of all the `if`s elsewhere. If we do that, then this isn’t a bug.